subsys/usb/class/cdc_acm.c

```diff
@@ -239,6 +243,13 @@ int cdc_acm_class_handle_req(struct usb_setup_packet *pSetup,
 		dev_data->line_state = (u8_t)sys_le16_to_cpu(pSetup->wValue);
 		LOG_DBG("CDC_SET_CONTROL_LINE_STATE 0x%x",
 			dev_data->line_state);
+#ifdef CONFIG_ARDUINO_LIKE_UPGRADE
+		if(sys_le32_to_cpu(dev_data->line_coding.dwDTERate) == 1200 && 
```

Comment:
> It also looks naive to me to reboot to bootloader based on baud rate setting.

I agree but that is that's mechanism those Backward-compatible boards use.

> I suggest to use non hand-free update procedure.

It's going to remain the default approach. I hid this code behind an `#ifdef` because I wanted there to be at least an _option_ to enable it and thus have the boards remain compatible with Arduino IDE should the user wish to do so.

> There should not be any platform/bootloader/update-technology code in the CDC ACM implementation.

Where do you suggest to put this then in an absence of a way to provide an external class_hande callback?

Reply:
> Where do you suggest to put this then in an absence of a way to provide an external class_hande callback?

Well I am also somewhat surprised about your arduino like approach. Regardless #ifdef or something else, it is wrong. You can not launch a bootloader from CDC ACM class implementation. Also, there will not be any callback for that in CDC ACM class implementation. The interface for the application is UART API. In the CDC ACM sample, you can find how to [obtain the baudrate, UART_LINE_CTRL_BAUD_RATE](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/subsys/usb/cdc_acm/src/main.c#L121). If it can not work for you then there is no other way.

Reply:
I know that doing it with the UART API would be cleaner abstraction-wise but because changes in DTR and the likes have to be polled manually, instead of having a callback, doing it this way would require a nasty spinning thread which only keeps polling the line.

> Also, there will not be any callback for that in CDC ACM class implementation.

There will not be any callback for what? The class changing? The DTR line? It's not clear the way you wrote it.

Reply:
>  know that doing it with the UART API would be cleaner abstraction-wise but because changes in DTR and the likes have to be polled manually, instead of having a callback, doing it this way would require a nasty spinning thread which only keeps polling the line.

You could use workqueue thread and timer for that. But it is your app and has nothing to do with CDC ACM implementation (or  subsys/power/reboot.c) or Zephyr RTOS self. And there are not _many_  boards but just few. What is the plan in general, to include USB device stack and ACM class per default? So if one would flash `samples/hello_world`, how to recover to bootloader? What you doing here is crap, stop it.



> There will not be any callback for what? The class changing? The DTR line? It's not clear the way you wrote it.

No.

Reply:
Look, I don't really care if this is in Zephyr or not. Just thought it might be useful for other people who would like to make their boards retain Arduino-IDE compatibility. Your responses sound like you are either not understanding what I'm asking, or are being intentionally dismissive and generally unpleasant.